### PR TITLE
Separate and expand information on testing for newcomers

### PR DESCRIPTION
**Pull request by apostasie**
It took me a little while to figure things out wrt testing.

This PR consolidates the information into a separate document and expands on it a bit.

I also added a new section in that doc giving some (hopefully useful) information about a few testing gotchas, parallelization, etc.

Let me know your thoughts and if this is helpful.

### CODE REVIEW

File: TESTING.md
Comment (AkihiroSuda) [inline TESTING.md:1]: Should this be in `docs/` ?

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# About testing nerdctl
```
Comment (AkihiroSuda) [inline TESTING.md:1]: "About" is probably not needed

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+### Or test in a container
+
+```bash
+docker build -t test-integration --target test-integration .
+docker run -t --rm --privileged test-integration
+```
+
+### Principles
+
+#### All tests shall be parallelized
```
Comment (AkihiroSuda) [inline TESTING.md:45]: (with best-effort)

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+Run `go test -v ./pkg/...`
+
+## Integration testing
+
+### TL;DR
+
+Be sure to first `make && sudo make install`
+
+```bash
+# Test all with nerdctl
```
Comment (AkihiroSuda) [inline TESTING.md:23]:
```suggestion
# Test all with nerdctl (rootless)
```

**Comment by apostasie** [comment]
Comments integrated.
Thanks @AkihiroSuda !

File: docs/TESTING.md
Comment (AkihiroSuda) [inline docs/TESTING.md:1]: Any reason to use UPPERCASE for the filename here?

Comment (apostasie) [reply docs/TESTING.md:1]: I guess when it was in the root folder - aligning with README and SECURITY.

Will lowercase though now that it is in docs.

Thanks.

Comment (apostasie) [reply docs/TESTING.md:1]: Done.

File: README.md [hunk 4]
```diff
@@ -249,32 +249,9 @@ See the header of [`go.mod`](./go.mod) for the minimum supported version of Go.
...
-```
-
-To run a single integration test (in this case, `image_inspect_test`):
-
-`go test -exec sudo -v ./cmd/nerdctl/main_test.go ./cmd/nerdctl/image_inspect_test.go `
-
-#### Running integration test suite against Docker
-
-Run `go test -exec sudo -v ./cmd/nerdctl/... -args -test.target=docker` to ensure that the test suite is compatible with Docker.
+See [testing nerdctl](docs/TESTING.md).
```
Comment (AkihiroSuda) [inline README.md:254]:
```suggestion
See [testing nerdctl](docs/testing.md).
```

Comment (apostasie) [reply README.md:254]: Damn... thanks!

**Comment by apostasie** [comment]
@AkihiroSuda hopefully capitalization should be fine this time........

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**